### PR TITLE
Resolve out-of-bounds accesses with dedicated server code [ASAN]

### PR DESCRIPTION
**Pull request by jengelh**
## Pull Request Type

- [x] Runtime changes
  - [ ] Network changes
  - [x] Other changes

### Related Issues

Fixes some of the issues plaguing #571.

### Checklist

- [x] I have partially tested my changes locally and verified that they work as intended.
- [x] I have reviewed the changes to ensure they do not introduce any unnecessary complexity or duplicate code.
- [x] I understand that by submitting this pull request, I am agreeing to license my contributions under the project's license.

### CODE REVIEW

**Comment by Lgt2x** [comment]
Possible concurrent changes with #579 published 6 minutes apart, needs sync with @winterheart :)

**Comment by jengelh** [comment]
In my defense, the introductory comment “What's the content...” effectively was a self-assign before the assigner field was actually set ;-)

**Comment by winterheart** [comment]
Let's just change `server_config_name` to std::filesystem::path and add path existence of it, and we'll be fine.

**Comment by jengelh** [comment]
Sounds very sensible.

File: Descent3/dedicated_server.cpp [hunk 1]
```diff
@@ -328,16 +328,16 @@ int LoadServerConfigFile() {
...
   } else {
     PrintDedicatedMessage(TXT_DS_MISSINGCONFIG);
     PrintDedicatedMessage("\n");
     return 0;
   }
 
   //	open file
-  if (!inf.Open(Netgame.server_config_name, "[server config file]", DedicatedServerLex)) {
-    PrintDedicatedMessage(TXT_DS_BADCONFIG, Netgame.server_config_name);
+  if (!inf.Open(Netgame.server_config_name.u8string().c_str(), "[server config file]", DedicatedServerLex)) {
```
Comment (winterheart) [inline Descent3/dedicated_server.cpp:339]: inf.Open() accepts std::fs::path, no need converting to char* here.

**Comment by winterheart** [comment]
Tested on prepared environment, fixes #571.

**Pull request merged**